### PR TITLE
Update Raft for Final Chapter

### PR DESCRIPTION
**Pull request by SunnyBat**
* Add new locations from new islands
* Add characters as location checks
* Victory condition is now the end of the final story island
* Add island distance modification option
* Add crafting table cost option

### CODE REVIEW

File: worlds/raft/items.json [hunk 1]
```diff
@@ -7,335 +7,460 @@
...
+        "name": "Advanced medium crop plot"
+    },
+    {
+        "id": 47027,
+        "progression": false,
+        "name": "Advanced large crop plot"
+    },
+    {
+        "id": 47028,
+        "progression": false,
```
Comment (Alchav) [inline worlds/raft/items.json:139]: Some items have incorrect progression flags, which is causing the unit test failure. I sent a PR directly to your repository to change some items which has fixed the unit test failure but should be double checked

File: worlds/raft/locations.json [hunk 2]
```diff
@@ -1,874 +1,1064 @@
...
-            "Thatch",
-            "Feather",
-            "Nail"
+            "MetalIngot"
         ]
     },
     {
         "id": 48008,
-        "name": "Smelter",
+        "name": "Shear",
```
Comment (Alchav) [inline worlds/raft/locations.json:70]: The item name was changed from Shear to Shears, but the location is still just Shear, is this intentional?

**Review by Alchav — CHANGES_REQUESTED** [review]

File: worlds/raft/items.json (hunk 1 above)
Comment (SunnyBat) [reply worlds/raft/items.json]: Wish I'd seen this before I spent the time debugging this LOL

File: worlds/raft/locations.json (hunk 2 above)
Comment (SunnyBat) [reply worlds/raft/locations.json:70]: Good catch. I think this got caught up in the mix of updating for the Final Chapter. Reverting to Shear to stay consistent with previous iterations.

File: worlds/raft/items.json (hunk 1 above)
Comment (Alchav) [reply worlds/raft/items.json]: I had thought about pinging you on discord about it, now I wish I had. Oh well

**Comment by Berserker66** [comment]
Shear and Bow&Arrow are no longer researches in 1.0?

**Comment by SunnyBat** [comment]
Those rules in particular are not necessary, though the items+locations themselves are still present. The Shears were never used in the first place. The Bow+Arrow wasn't actually necessary to progress (you can throw stones instead of firing an arrow to hit a lever and progress through the story island), so it was removed as a requirement.

File: worlds/raft/items.json [hunk 3]
```diff
@@ -7,335 +7,460 @@
...
         "id": 47011,
         "progression": false,
-        "name": "Bucket"
+        "name": "Electric zipline tool"
     },
     {
         "id": 47012,
         "progression": false,
-        "name": "Healing salve"
+        "name": "Empty bottle"
```
Comment (Alchav) [inline worlds/raft/items.json:60]: As mentioned on discord, I believe this should be progression as `raft_can_craft_plasticBottle` requires this item

Comment (SunnyBat) [reply worlds/raft/items.json:60]: Good catch again :)

**Review by Alchav — APPROVED** [review]

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**